crates/biome_js_analyze/src/aria_analyzers/nursery/use_valid_aria_role.rs

```diff
@@ -0,0 +1,117 @@
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="datepicker"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="range"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role=""></div>
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    ///   <Foo role="foo"></Foo>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <>
+    ///   <div role="button"></div>
+    ///   <div role={role}></div>
+    ///   <div></div>
+    /// </>
+    /// ```
+    ///
+    /// ## Accessibility guidelines
+    /// - [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+    ///
+    /// ## Resources
+    /// - [Chrome Audit Rules, AX_ARIA_01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01)
+    /// - [DPUB-ARIA roles](https://www.w3.org/TR/dpub-aria-1.0/)
+    /// - [MDN: Using ARIA: Roles, states, and properties](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques)
+    ///
+    pub(crate) UseValidAriaRole {
+        version: "next",
+        name: "useValidAriaRole",
+        recommended: false,
+    }
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct UseValidAriaRoleOptions {
+    allowed_invalid_roles: Vec<String>,
+    ignore_non_dom: bool,
+}
+
+impl Rule for UseValidAriaRole {
+    type Query = Aria<AnyJsxElement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseValidAriaRoleOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let options = ctx.options();
+        let aria_roles = ctx.aria_roles();
+
+        let ignore_non_dom = options.ignore_non_dom;
+        let allowed_invalid_roles = &options.allowed_invalid_roles;
+
+        if ignore_non_dom && node.is_custom_component() {
+            return None;
+        }
+
+        let role_attribute = node.find_attribute_by_name("role")?;
+
+        let role_attribute_static_value = role_attribute.as_static_value()?;
+        let role_attribute_value = role_attribute_static_value.text();
+        let mut role_attribute_value = role_attribute_value.split(' ');
+
+        let is_valid = role_attribute_value.all(|val| {
+            let role_data = aria_roles.get_role(val);
+            allowed_invalid_roles.contains(&val.to_string()) || role_data.is_some()
+        });
+
+        if is_valid {
+            return None;
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Enforce that elements with ARIA roles must use a valid, non-abstract ARIA role."
```

Comment:
This is the phrase you want in the documentation, not here. Here, you want to have something like "The role **<ROLE_NAME>** doesn't exist", or `"The role **<ROLE_NAME> is abstract and can't be applied.*